### ISSUE TITLE
InstantRescue App

### ISSUE THREAD

**Issue by husder** [HydroCommunity/Community-Brainstorming#28]
I have an idea that would save thousands of life especially here in the Philippines. 

When there are accidents, calamities or someone needs help. Here in Philippines it takes longer time  for the rescuer/ambulance/police to respond and saves the life of the victim or person in need. The patient maybe indanger because no one can help immediately.

Factors of delayed:
1. The ambulance or the doctor will arrive on the incidents area late and sometimes it will cost the life of the patient because of traffic.
2. The hospital or the ambulance is too far from the area where the patient needs help
3. Due to panic the family members sometimes don't know where to call and what number to call to ask for help.

So how InstantRescue App will help the patient?
Solution:
1. In our app there's a list of volunteers that are available anytime to help. These volunteers should undergo training or must have a certificate to show inorder to become a certified rescuer.  The common volunteer that are automatically accepted are doctors, nurses and those who work in nonprofit agencies like WHO and Red Cross that can show some proof about their trainings example handling CPR and/or can perform first aid treatment to the patient. Let say, in our country in the Philippines people will download the app and we get tons of registered rescuer, in that way when the rescuer can arrive earlier there's a big chance that he/she can save the life.

2. Second, if the patient needs to be brought to the Emergency Room asap due to the condition, the rescuer can apply the first aid treatment and I suggest that the registered volunteer/rescuer should have a motorcycle where he/she can use it if someone needs some help to avoid traffic. Should be accompanied by someone like a family member so that it would not be hard for the driver.

3. If you are in danger example someone is trying to kill you, you can use the app, maybe there's an emergency button and you just have a selfie and the app will automatically locate the area using google map. So the police can easily locate you.

4. If most of the Filipinos have this app, it can really save more lives. They can also report every accidents/incidents in their area and the police or the nearest volunteer will be notify. I don't know how it can be done but maybe we can get an idea with a Grab app, wherein the drivers can view or notified if someone booked on the app. So in our case maybe they can also check if someone ask for help in the area near you. Everything is possible even if you are in the top of the mountain, I'm sure their's someone who is willing to help you.

5. This app can also help a patient in finding blood donors, example scenario; I can relate to this because my aunt died last 2 months ago. Although she has a cervical cancer but when she was brought to the hospital due to excessive bleeding, meaning she loss too much blood and she need 5 bags of type B blood but the nurses said they don't have available blood, that's why it takes several hours before she received blood transfusion, it happen early in the morning and the blood is available in the afternoon around 2pm bacause my uncle can't find a donor for blood easily. So my aunt died at 8pm, I think if she received the blood transfusion early she can still survive. So let's go back to the app, what it does is that, theirs also a section wherein you can find also volunteers who regularly donates blood so we can also ask help from the Red Cross about the records (optional). Any blood donors with your blood type near you can easily donate blood for you.

6. Actually, this app is very useful in any accidents, incidents that may happen or occur in your life you can ask help from it. Even if your dog was dying you can still ask help from there.In the registration, we need to categorize what type of help they are willing to volunteer or they are expert at. 

7. Optional function for fund raising for a treatment etc. and a function that we accept donations.

Suggestion for the volunteers:
1. Maybe they must have the first aid kit, that can be use whenever their's an accident. I don't know if they should provide it or I don't know how we can provide it to them.
2. They must have a proper trainings, so we need assistance also from health organizations that provide this training for free.
3. Must have a motorcycle so that they can use it when needed. It's easier for them to respond because they can avoid the traffic.

Why I prefer Blockchain?
I'm a believer of blockchain technology and I know the potential of using it, I want every transactions to be transparent and secure. Everyone who wanted to volunteer can't pass without a proof of work or a certification/license to show. Blockchain technology also is very suitable for the function of blood donor, it is easier for the patient to find donors if we have the records.

I am open for any suggestion how to make it work, my initial plan is after we can create a prototype maybe we can propose it to the government and let them check the importance and what help it can bring to everyone. If they will support it then it's better, it is easier to bring it to the public because they will help us introduce it to the public if they don't like we can still share it to the public.

So the challenge here is how we can convince the public to download the app and invite volunteers to register with valid documents and proof of work. 

Best scenario is if we can get assistance from Hospitals, Red Cross, NBI and Police and health institutions wherein the patient can receive the help instantly.

Optional: This project can be tokenized and we will try it first in the Philippines and expand if other countries don't have this kind of concept, they can also use it.

I hope you will like it, I am open for any suggestion as long as it can improve the concept.

**Comment by MasterSensei** [comment]
Awesome idea! This is something that definitely could be created on top of ERC-1484 Hydro Snowflake to prove and attest to their medical credentials. You could also store things like medical records through Hydro Ice.

**Comment by ebunayo** [comment]
A very great idea this is, how about a team of Philippines come together to develop an app that serves this function, incentivize it in such a way that volunteers get duly and fully rewarded. Of course the app will be integrated with snowflakes and also raindrop and also developers could contribute to this from all over the world and work with a local team that knows and understands the terrain

**Comment by husder** [comment]
> Awesome idea! This is something that definitely could be created on top of ERC-1484 Hydro Snowflake to prove and attest to their medical credentials. You could also store things like medical records through Hydro Ice.
Thank you bro, I hope more blockchain developers will work on this project so that we could save more lives. :)

**Comment by husder** [comment]
> A very great idea this is, how about a team of Philippines come together to develop an app that serves this function, incentivize it in such a way that volunteers get duly and fully rewarded. Of course the app will be integrated with snowflakes and also raindrop and also developers could contribute to this from all over the world and work with a local team that knows and understands the terrain

Good idea to give incentivize the volunteers. We are building a community of volunteers that are competent enough to save more lives.

**Comment by ebunayo** [comment]
I think there might be a need for a proper documentation identifying those that will be beneficiaries, partnering hospitals, aid and donor agencies to fund the incentives as motivations.

There is also need for extensive research and collaborations.

**Comment by emilysw02** [comment]
This is such an amazing idea and one I've wished existed so many times but never so neatly packaged.  Imagine how many lives it could save when people live in rural areas even in north america and an ambulance takes an hour or more, or in apartments where tons of people live in close quarters and a bunch have their first aid but someone in need of cpr is waiting 5 minutes for an ambulance.
